vagrant/vagrant-setup.sh

```diff
@@ -6,8 +6,29 @@
 
 VAGRANT_PKG_URL="https://releases.hashicorp.com/vagrant/2.2.4/vagrant_2.2.4_x86_64.rpm"
 
-set -e -u
 set -o pipefail
+set -u
+
+# Set up nested KVM
+KVM_MODULE_NAME="$(lsmod | grep -m1 -Eo '(kvm_intel|kvm_amd)')"
```

Comment:
I suggest keeping the `set -e` active.

Just use a `|| :` at the end of this line to prevent the error from here:

```sh
# Set up nested KVM, if already running inside KVM.
KVM_MODULE_NAME="$(lsmod | grep -m1 -Eo '(kvm_intel|kvm_amd)')" || :
```

You can do the same if there are other lines that might trigger an error due to non-zero exit that's expected.

Reply:
The `set -e` is still active, I simply moved it just after this function. The idea here is that the nested KVM stuff is not *crucial* for the testing itself - it can run fine without it. I agree that having it enable from the start it way nicer though. I'll try to rewrite it a little bit.